modules/dnn/src/tflite/tflite_importer.cpp

```diff
@@ -0,0 +1,582 @@
+    if (opcode == "PRELU") {
+        layerParams.type = "PReLU";
+    } else if (opcode == "RELU") {
+        layerParams.type = "ReLU";
+    } else if (opcode == "ADD") {
+        layerParams.type = "Eltwise";
+        layerParams.set("operation", "sum");
+    } else if (opcode == "MUL") {
+        layerParams.type = "Eltwise";
+        layerParams.set("operation", "prod");
+    } else if (opcode == "HARD_SWISH") {
+        layerParams.type = "HardSwish";
+    } else if (opcode == "LOGISTIC") {
+        layerParams.type = "Sigmoid";
+    } else {
+        CV_Error(Error::StsNotImplemented, "Unknown eltwise operator opcode: " + opcode);
+    }
+}
+
+void TFLiteImporter::parsePooling(const Operator& op, const std::string& opcode, LayerParams& layerParams) {
+    layerParams.type = "Pooling";
+
+    auto options = reinterpret_cast<const Pool2DOptions*>(op.builtin_options());
+    layerParams.set("pad_mode", EnumNamePadding(options->padding()));
+    layerParams.set("stride_w", options->stride_w());
+    layerParams.set("stride_h", options->stride_h());
+    layerParams.set("kernel_w", options->filter_width());
+    layerParams.set("kernel_h", options->filter_height());
+    if (opcode == "MAX_POOL_2D")
+        layerParams.set("pool", "max");
+    else if (opcode == "AVERAGE_POOL_2D")
+        layerParams.set("pool", "ave");
+    else
+        CV_Error(Error::StsNotImplemented, "Pool type selection for " + opcode);
+}
+
+void TFLiteImporter::parsePoolingWithArgmax(const Operator& op, const std::string& opcode, LayerParams& layerParams) {
+    layerParams.type = "Pooling";
+
+    CV_CheckLE(op.custom_options()->size(), sizeof(TfLitePoolParams), "");
+    const auto* params = reinterpret_cast<const TfLitePoolParams*>(op.custom_options()->Data());
+    if (params->padding != kTfLitePaddingUnknown)
+        layerParams.set("pad_mode", params->padding == kTfLitePaddingSame ? "SAME" : "VALID");
+    layerParams.set("stride_w", params->stride_width);
+    layerParams.set("stride_h", params->stride_height);
+    layerParams.set("kernel_w", params->filter_width);
+    layerParams.set("kernel_h", params->filter_height);
+    layerParams.set("pool", "max");
+}
+
+void TFLiteImporter::parseUnpooling(const Operator& op, const std::string& opcode, LayerParams& layerParams) {
+    layerParams.type = "MaxUnpool";
+
+    CV_CheckLE(op.custom_options()->size(), sizeof(TfLitePoolParams), "");
+    const auto* params = reinterpret_cast<const TfLitePoolParams*>(op.custom_options()->Data());
+    layerParams.set("pool_stride_w", params->stride_width);
+    layerParams.set("pool_stride_h", params->stride_height);
+    layerParams.set("pool_k_w", params->filter_width);
+    layerParams.set("pool_k_h", params->filter_height);
+    layerParams.set("pool_pad_w", 0);
+    layerParams.set("pool_pad_h", 0);
+}
+
+void TFLiteImporter::parseReshape(const Operator& op, const std::string& opcode, LayerParams& layerParams) {
```

Comment:
Not familiar with TFLite models but do we need to consider constant folding for some of the operators?

Reply:
Not sure that it must be done at import step due DNN has own fusion at the model initialization. If you mean something similar to TensorFlow/ONNX graph ops fusion, TFLite  basic layers (see the schema) are simple enough and mapped 1:1 to OpenCV layers. At least according the models I tried from Mediapipe.

Reply:
> If you mean something similar to TensorFlow/ONNX graph ops fusion

Yes, I meant this. Just wonder whether there are similar cases in TFLite like "[Input]->Identity->..." where the Identity can be eliminated in ONNX. But still this can be a further optimization if we comes across these cases.

Reply:
Well, yes, there are builtin operations which probably can be a part of single layers. I had an idea to make a something like general rules between TensorFlow and TFLite importers for NHWC-NCHW layout tracking and subgraph fusion. However, it may be also done later with more TFLite models examples.

<details>
<summary>From schema.fbs</summary>

```
enum BuiltinOperator : int32 {
  ADD = 0,
  AVERAGE_POOL_2D = 1,
  CONCATENATION = 2,
  CONV_2D = 3,
  DEPTHWISE_CONV_2D = 4,
  DEPTH_TO_SPACE = 5,
  DEQUANTIZE = 6,
  EMBEDDING_LOOKUP = 7,
  FLOOR = 8,
  FULLY_CONNECTED = 9,
  HASHTABLE_LOOKUP = 10,
  L2_NORMALIZATION = 11,
  L2_POOL_2D = 12,
  LOCAL_RESPONSE_NORMALIZATION = 13,
  LOGISTIC = 14,
  LSH_PROJECTION = 15,
  LSTM = 16,
  MAX_POOL_2D = 17,
  MUL = 18,
  RELU = 19,
  // NOTE(aselle): RELU_N1_TO_1 used to be called RELU1, but it was renamed
  // since different model developers use RELU1 in different ways. Never
  // create another op called RELU1.
  RELU_N1_TO_1 = 20,
  RELU6 = 21,
  RESHAPE = 22,
  RESIZE_BILINEAR = 23,
  RNN = 24,
  SOFTMAX = 25,
  SPACE_TO_DEPTH = 26,
  SVDF = 27,
  TANH = 28,
  CONCAT_EMBEDDINGS = 29,
  SKIP_GRAM = 30,
  CALL = 31,
  CUSTOM = 32,
  EMBEDDING_LOOKUP_SPARSE = 33,
  PAD = 34,
  UNIDIRECTIONAL_SEQUENCE_RNN = 35,
  GATHER = 36,
  BATCH_TO_SPACE_ND = 37,
  SPACE_TO_BATCH_ND = 38,
  TRANSPOSE = 39,
  MEAN = 40,
  SUB = 41,
  DIV = 42,
  SQUEEZE = 43,
  UNIDIRECTIONAL_SEQUENCE_LSTM = 44,
  STRIDED_SLICE = 45,
  BIDIRECTIONAL_SEQUENCE_RNN = 46,
  EXP = 47,
  TOPK_V2 = 48,
  SPLIT = 49,
  LOG_SOFTMAX = 50,
  // DELEGATE is a special op type for the operations which are delegated to
  // other backends.
  // WARNING: Experimental interface, subject to change
  DELEGATE = 51,
  BIDIRECTIONAL_SEQUENCE_LSTM = 52,
  CAST = 53,
  PRELU = 54,
  MAXIMUM = 55,
  ARG_MAX = 56,
  MINIMUM = 57,
  LESS = 58,
  NEG = 59,
  PADV2 = 60,
  GREATER = 61,
  GREATER_EQUAL = 62,
  LESS_EQUAL = 63,
  SELECT = 64,
  SLICE = 65,
  SIN = 66,
  TRANSPOSE_CONV = 67,
  SPARSE_TO_DENSE = 68,
  TILE = 69,
  EXPAND_DIMS = 70,
  EQUAL = 71,
  NOT_EQUAL = 72,
  LOG = 73,
  SUM = 74,
  SQRT = 75,
  RSQRT = 76,
  SHAPE = 77,
  POW = 78,
  ARG_MIN = 79,
  FAKE_QUANT = 80,
  REDUCE_PROD = 81,
  REDUCE_MAX = 82,
  PACK = 83,
  LOGICAL_OR = 84,
  ONE_HOT = 85,
  LOGICAL_AND = 86,
  LOGICAL_NOT = 87,
  UNPACK = 88,
  REDUCE_MIN = 89,
  FLOOR_DIV = 90,
  REDUCE_ANY = 91,
  SQUARE = 92,
  ZEROS_LIKE = 93,
  FILL = 94,
  FLOOR_MOD = 95,
  RANGE = 96,
  RESIZE_NEAREST_NEIGHBOR = 97,
  LEAKY_RELU = 98,
  SQUARED_DIFFERENCE = 99,
  MIRROR_PAD = 100,
  ABS = 101,
  SPLIT_V = 102,
  UNIQUE = 103,
  CEIL = 104,
  REVERSE_V2 = 105,
  ADD_N = 106,
  GATHER_ND = 107,
  COS = 108,
  WHERE = 109,
  RANK = 110,
  ELU = 111,
  REVERSE_SEQUENCE = 112,
  MATRIX_DIAG = 113,
  QUANTIZE = 114,
  MATRIX_SET_DIAG = 115,
  ROUND = 116,
  HARD_SWISH = 117,
  IF = 118,
  WHILE = 119,
  NON_MAX_SUPPRESSION_V4 = 120,
  NON_MAX_SUPPRESSION_V5 = 121,
  SCATTER_ND = 122,
  SELECT_V2 = 123,
  DENSIFY = 124,
  SEGMENT_SUM = 125,
  BATCH_MATMUL = 126,
  PLACEHOLDER_FOR_GREATER_OP_CODES = 127,
  CUMSUM = 128,
  CALL_ONCE = 129,
  BROADCAST_TO = 130,
  RFFT2D = 131,
  CONV_3D = 132,
  IMAG=133,
  REAL=134,
  COMPLEX_ABS=135,
  HASHTABLE = 136,
  HASHTABLE_FIND = 137,
  HASHTABLE_IMPORT = 138,
  HASHTABLE_SIZE = 139,
  REDUCE_ALL = 140,
  CONV_3D_TRANSPOSE = 141,
  VAR_HANDLE = 142,
  READ_VARIABLE = 143,
  ASSIGN_VARIABLE = 144,
  BROADCAST_ARGS = 145,
  RANDOM_STANDARD_NORMAL = 146,
  BUCKETIZE = 147,
  RANDOM_UNIFORM = 148,
  MULTINOMIAL = 149,
  GELU = 150,
  DYNAMIC_UPDATE_SLICE = 151,
  RELU_0_TO_1 = 152,
  UNSORTED_SEGMENT_PROD = 153,
  UNSORTED_SEGMENT_MAX = 154,
  UNSORTED_SEGMENT_SUM = 155,
  ATAN2 = 156,
  UNSORTED_SEGMENT_MIN = 157,
  SIGN = 158
}
```

</details>